### PR TITLE
Pep8

### PR DESCRIPTION
**Pull request by thisfred**
I added a few commands to the Makefile to help with (de)linting, and I did an initial run of `make autopep8` to clean up the style/formatting issues that were automatically fixable. After that I fixed some more of the reported lint issues by hand.

### CODE REVIEW

File: ensime_shared/ensime.py [hunk 1]
```diff
@@ -217,12 +213,6 @@ def set_cursor(self, row, col):
         self.log("set_cursor: {}".format((row, col)))
         self.vim.current.window.cursor = (row, col)
 
-    def set_cursor(self, char_counter):
-        """Set cursor at nth char in a buffer."""
-        self.log("set_cursor: {}".format((row, col)))
-        cmd = commands["go_to_char"].format(char_counter + 1)
-        self.vim.command(cmd)
```
Comment (thisfred) [inline ensime_shared/ensime.py]: this is a redefinition of an earlier method and has a syntax error in it, so I assumed the other version was the currently used one.

File: ensime_shared/ensime.py [hunk 2]
```diff
@@ -12,13 +12,8 @@
 from subprocess import Popen, PIPE
 
 import json
-import os
-import re
-import logging
 import time
 import datetime
-import inspect
-import unicodedata
```
Comment (thisfred) [inline ensime_shared/ensime.py]: unused/duplicate imports removed

File: ensime_shared/ensime.py [hunk 3]
```diff
@@ -628,9 +618,6 @@ def send_refactor_request(self, ref_type, ref_params, ref_options):
 
     def apply_refactor(self, call_id, payload):
         """Apply a refactor depending on its type."""
-        refactor_id = payload["procedureId"]
-        current_file = self.refactorings.get(refactor_id)
```
Comment (thisfred) [inline ensime_shared/ensime.py]: unused variables removed

File: ensime_shared/launcher.py [hunk 4]
```diff
@@ -5,9 +5,10 @@
 import re
 
 from ensime_shared.util import Util
-from ensime_shared.config import gconfig, feedback
```
Comment (thisfred) [inline ensime_shared/launcher.py]: removed unused imports

File: ensime_shared/launcher.py [hunk 5]
```diff
@@ -85,9 +90,16 @@ def load_classpath(self, scala_version, java_home):
         classpath_file = os.path.join(project_dir, "classpath")
         if not os.path.exists(classpath_file):
             self.generate_classpath(scala_version, classpath_file)
-        return "{}:{}/lib/tools.jar".format(Util.read_file(classpath_file), java_home)
-
-    def start_process(self, conf_path, classpath, cache_dir, java_home, java_flags):
+        return "{}:{}/lib/tools.jar".format(
+            Util.read_file(classpath_file), java_home)
+
+    def start_process(
```
Comment (jvican) [inline ensime_shared/launcher.py]: i would change the declaration of this method, maybe two lines? line per argument is very cumbersome

File: ensime_shared/launcher.py [hunk 6]
```diff
@@ -113,14 +127,21 @@ def generate_classpath(self, scala_version, classpath_file):
...
-        Util.write_file(os.path.join(project_dir, "build.sbt"), self.build_sbt(scala_version, classpath_file))
-        Util.write_file(os.path.join(project_dir, "project", "build.properties"),
-                "sbt.version={}".format(self.sbt_version))
-        log_file = os.path.join(project_dir, "build.log")
-        log = open(log_file, 'w')
-        null = open("/dev/null", "r")
-        # see https://github.com/ensime/ensime-vim/issues/29 on why we use this method
-        self.vim.command("!(cd {};sbt -Dsbt.log.noformat=true -batch saveClasspath)".format(project_dir))
+        Util.write_file(
+            os.path.join(project_dir, "build.sbt"),
```
Comment (jvican) [inline ensime_shared/launcher.py]: same here, wouldn't be better to use '\' and split this in two lines?

**Comment by jvican** [comment]
LGTM

**Comment by jvican** [comment]
I'm gonna merge this as soon as you rebase on top of the last PR that got merged and check the changes @thisfred, thanks!

File: ensime_shared/launcher.py (hunk 5 above)
Comment (thisfred) [inline ensime_shared/launcher.py]: I will change that.

File: ensime_shared/launcher.py (hunk 6 above)
Comment (thisfred) [inline ensime_shared/launcher.py]: Not 100% what your preferred solution is. (`\` is only needed in Python if there are no enclosing parentheses, brackets or braces, but again, I'm not entirely sure what you're proposing to change.)

Comment (jvican) [inline ensime_shared/launcher.py]: True, thanks for pointing that out. What I mean is that there's no need to put each parameter in every line (two of them could fit and comply with the 80chars rule). I don't mind in this case, but I think that it's very ugly in the case of the declaration of a function (as you did in other change). But anyway, leave it like it is, it was just a matter of style :smile:

Comment (thisfred) [inline ensime_shared/launcher.py]: Thanks, I changed it to be only two lines, which I agree does look better.

**Comment by jvican** [comment]
Ready for merge @thisfred ?

**Comment by thisfred** [comment]
@jvican: yep, this one's ready

**Comment by jvican** [comment]
Please, squash commits and split them into conceptual units (one commit would be the Makefile and the other one the PEP8 compliance. BTW, thanks for the contribution!

**Comment by thisfred** [comment]
Squashed into two logical commits. Thanks for helping me get it up to standards!

**Comment by jvican** [comment]
Thanks to you @thisfred, if you are interested in contributing more you can check some open tickets on the issues page (e.g. refactorings need to be finished on the client side) :dancer:

**Comment by thisfred** [comment]
Thank you! I will definitely start looking at open issues next.

**Pull request merged**